File: internal/app/paths.go
package app

import (
	"os"
	"path/filepath"
)

func DataDir() (string, error) {
	dir, err := os.UserConfigDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(dir, "nav"), nil
}

func DatabasePath() (string, error) {
	dir, err := DataDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(dir, "nav.db"), nil
}

File: src/paths.ts
import { join } from "node:path";

export function dataDirectory(environment: NodeJS.ProcessEnv = process.env): string {
  if (process.platform === "win32") {
    const root = environment.LOCALAPPDATA;
    if (!root) throw new Error("LOCALAPPDATA is not set");
    return join(root, "Nav");
  }
  if (process.platform === "darwin") {
    const home = environment.HOME;
    if (!home) throw new Error("HOME is not set");
    return join(home, "Library", "Application Support", "nav");
  }
  const root = environment.XDG_STATE_HOME;
  if (root) return join(root, "nav");
  const home = environment.HOME;
  if (!home) throw new Error("HOME is not set");
  return join(home, ".local", "state", "nav");
}

export function databasePath(environment: NodeJS.ProcessEnv = process.env): string {
  return join(dataDirectory(environment), "nav.db");
}
